### PR TITLE
1. Removed overload == and != operators for matrix from header and te…

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
…st file

2. Replaced with isClose function that takes the tolerance as a parameter
3. Added a floating point comparison function to a new file and added a test for it

### CODE REVIEW

**Comment by JoshuaSBrown** [comment]
@JensWehner check if you think this is better.

File: include/votca/tools/matrix.h [hunk 1]
```diff
@@ -17,10 +17,10 @@
 
 #ifndef _VOTCA_TOOLS_MAT_H
 #define _VOTCA_TOOLS_MAT_H
-
 #include "ostream"
-#include "types.h"
-#include "vec.h"
+#include <votca/tools/floatingpointcomparison.h>
+#include <votca/tools/types.h>
+#include <votca/tools/vec.h>
```
Comment (junghans) [inline include/votca/tools/matrix.h]: For headers in the same package, please use relative path (as it was before for vec.h)

Comment (JoshuaSBrown) [reply include/votca/tools/matrix.h]: Is this good practice or just because its been the convention?

Comment (junghans) [reply include/votca/tools/matrix.h]: Convention and consistency with other headers!

Comment (junghans) [reply include/votca/tools/matrix.h]: https://stackoverflow.com/questions/597318/what-are-the-benefits-of-a-relative-path-such-as-include-header-h-for-a-hea

I don't think it matters much, but consistency!

**Comment by JoshuaSBrown** [comment]
@JensWehner  Are you saying replace 
```inline bool operator==(const vec &v1, const vec &v2)
{
    return ((v1.getX()==v2.getX()) && (v1.getY()==v2.getY()) && (v1.getZ()==v2.getZ()));
}

inline bool operator!=(const vec &v1, const vec &v2)
{
    return ((v1.getX()!=v2.getX()) || (v1.getY()!=v2.getY()) || (v1.getZ()==v2.getZ()));
}

```
with a isClose function?

**Comment by JoshuaSBrown** [comment]
@JensWehner yes very nice! You  still didn't answer my question though.

**Comment by JensWehner** [comment]
What was your question?

**Comment by JensWehner** [comment]
yeah we should replace them, @onurcaylak  is working on that.

**Comment by JoshuaSBrown** [comment]
Should I leave this branch open for that or do you want to add that to this one?

**Comment by JensWehner** [comment]
you can close after merging.

Good job

**Comment by JensWehner** [comment]
@junghans can you force merge? I do not know why travis fails?

**Comment by junghans** [comment]
@JensWehner some API in gromacs master, should be fixed in votca/csg#244.

**Pull request merged**